File: 1341-split-a-string-in-balanced-strings/solution.java
class Solution {
    public int balancedStringSplit(String s) {
        int bal=0,count=0;

        for(char c:s.toCharArray()){
            if(c=='R'){
                bal++;
            }else{
                bal--;
            }

            if(bal==0) count++;
        }
        return count;
    }
}

File: 1341-split-a-string-in-balanced-strings/solution.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        bal=0
        count=0
        for i in s:
            if i=='L':
                bal+=1
            else:
                bal-=1
            
            if bal==0:
                count+=1
        return count
        

